### PR TITLE
Add 2 NLI models and update 1 coref model

### PR DESCRIPTION
**Pull request by ZhaofengWu**
This PR adds 2 new RoBERTa-based NLI models and update the BERT-based coref model to use (a) SpanBERT and (2) 2nd order inference and coarse-to-fine inference. I also worked on a new sentiment analysis model and that's already in the demo (#323).

This PR is based on [a branch of Dirk's fork](https://github.com/dirkgr/allennlp-demo/commits/FixEverything) (specifically [this commit](https://github.com/dirkgr/allennlp-demo/commit/3a404f65dbef1cb69ff5633c60ba6b502b15ddfa)) for the dependency on the allennlp-models repo. I tested that these models that I added/updated are working _locally_ with allennlp version 2a1b528f and allennlp-models version c8266073. By _locally_ I mean following the `README.md`.

**What needs to be done before being merged**
Since the Dirk's branch that I based on added the dependency on allennlp-models, and that many demo models are not yet compatible with allennlp-models, they need to be updated before this can be merged. There might be some rebasing work that needs to be done then.

### CODE REVIEW

**Comment by dirkgr** [comment]
Can you make this PR merge into my branch instead of into master?

File: app.py [hunk 1]
```diff
@@ -50,7 +50,9 @@
                               'fine-grained-named-entity-recognition',
                               'glove-sentiment-analysis',
                               'roberta-sentiment-analysis',
-                              'textual-entailment',
```
Comment (dirkgr) [inline app.py]: This removes `textual-entailment`?

Comment (ZhaofengWu) [reply app.py]: `textual-entailment` is now `elmo-snli`. Because now there're 3 models for entailment, I want more discerning names

Comment (dirkgr) [reply app.py]: Oh, I see. You just renamed it.

**Comment by ZhaofengWu** [comment]
I don't have write access to your repo but you can feel free to do that. It's on a branch `zhaofeng_models` on upstream.

**Comment by dirkgr** [comment]
You don't need write access to make a PR.

**Comment by ZhaofengWu** [comment]
Oh right, ok

**Comment by dirkgr** [comment]
We merged this via another branch.

**Comment by dirkgr** [comment]
If we didn't squash commits, GitHub would know this and not print scary warnings.

**Pull request closed without merging**